src/contribute-to-tidb/contribute-code.md

```diff
@@ -1,88 +1,53 @@
-Contributors can expect to get a first reaction from a committer within a few days after opening the issue. If an issue doesn’t get any attention, we recommend reaching out to the [TiDB Internals forum](https://internals.tidb.io/). Note that the community sometimes does not have the capacity to accept all incoming contributions.
+  - Do *not* fix more than one issue in the same commit (except, of course, if one code change fixes all of them).
+  - Do *not* do cosmetic changes to unrelated code in the same commit as some feature/bugfix.
 
-Once all requirements for the issue are met, one could assign somebody to the issue to work on it.
+## Waiting for review
 
-### 2. Implement your change
+To begin with, please be patient! There are many more people submitting pull requests than there are people capable of reviewing your pull request. Getting your pull request reviewed requires a reviewer to have the spare time and motivation to look at your pull request. If your pull request has not received any notice from reviewers (i.e., no comment made) for some time, you can ping the reviewers and asignees, or take it to [internal.tidb.io](https://internals.tidb.io) for more attention.
 
-Once you've been assigned to a GitHub issue, you may start to implement the required changes.
-
-Here are some further points to keep in mind while implementing:
-
-* Set up a TiDB development environment.
-* Follow the [Code Style and Quality Guide](code-style-and-quality-guide.md) of TiDB.
-* Take any discussions and requirements from the GitHub issue or design document into account.
-* Do not mix unrelated issues into one contribution.
-
-### 3. Open a Pull Request
-
-Considerations before opening a pull request:
-
-* Make sure that `make test` is passing on your changes to ensure that all checks pass, the code builds and that all tests pass.
-* Make sure your commit history adheres to the requirements.
-* Make sure your change is up to date with the base branch.
-
-Code changes in TiDB are reviewed and accepted through [GitHub pull requests](https://help.github.com/en/articles/creating-a-pull-request).
-
-GitHub will auto assign reviewers to the pull requests, but you can also request reviews by commenting `/cc @reviewer`.
-
-There is a separate guide on [how to review a pull request](review-a-pr.md), including our pull request review process. As a code author, you should prepare your pull request to meet all requirements.
-
-### 4. Merge change
-
-The code will be merged by a committer of TiDB once the review is finished. The related GitHub issue will be closed afterwards.
+When someone does manage to find the time to look at your pull request they will most likely make comments about how it can be improved (don’t worry, even committers/maintainers have their pull requests sent back to them for changes). It is then expected that you update your pull request to address these comments, and the review process will thus iterate until a satisfactory solution has emerged.
```

Comment:
```suggestion
When someone does manage to find the time to look at your pull request, they will most likely make comments about how it can be improved (don't worry, even committers/maintainers have their pull requests sent back to them for changes). It is then expected that you update your pull request to address these comments, and the review process will thus iterate until a satisfactory solution has emerged.
```